File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        boolean result = false;
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            if (set.contains(nums[i])) {
                result = true;
                break;
            }
            else {
                set.add(nums[i]);
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> someSet;

        for (int i = 0; i < nums.size(); i++) {
            someSet.insert(nums[i]);
        }

        //find if it has a predecessor OR if it has a successor
        int maxLen = 0;

        for (const auto& elem : someSet) {
            if (!someSet.contains(elem - 1)) {
                int count = 1;
                int val = 1;
                while (someSet.contains(elem + val)) {
                    count++;
                    val++;
                }
                if (count > maxLen) {
                    maxLen = count;
                }
            }
        }

        return maxLen;
    }
};
